tests/unit-tests/CombinationComplexCriteria.swift

```diff
@@ -0,0 +1,549 @@
+                             "comparatorType": "StartsWith",
+                             "value": "A",
+                             "fieldType": "string"
+                           }
+                         ]
+                       }
+                     },
+                     {
+                       "dataType": "user",
+                       "searchCombo": {
+                         "combinator": "And",
+                         "searchQueries": [
+                           {
+                             "dataType": "user",
+                             "field": "lastName",
+                             "comparatorType": "StartsWith",
+                             "value": "A",
+                             "fieldType": "string"
+                           }
+                         ]
+                       }
+                     }
+                   ]
+                 },
+                 {
+                   "combinator": "Or",
+                   "searchQueries": [
+                     {
+                       "dataType": "user",
+                       "searchCombo": {
+                         "combinator": "And",
+                         "searchQueries": [
+                           {
+                             "dataType": "user",
+                             "field": "firstName",
+                             "comparatorType": "StartsWith",
+                             "value": "C",
+                             "fieldType": "string"
+                           }
+                         ]
+                       }
+                     },
+                     {
+                       "dataType": "customEvent",
+                       "searchCombo": {
+                         "combinator": "And",
+                         "searchQueries": [
+                           {
+                             "dataType": "customEvent",
+                             "field": "animal-found.vaccinated",
+                             "comparatorType": "Equals",
+                             "value": "false",
+                             "fieldType": "boolean"
+                           }
+                         ]
+                       }
+                     },
+                     {
+                       "dataType": "customEvent",
+                       "searchCombo": {
+                         "combinator": "And",
+                         "searchQueries": [
+                           {
+                             "dataType": "customEvent",
+                             "field": "animal-found.count",
+                             "comparatorType": "LessThan",
+                             "value": "5",
+                             "fieldType": "long"
+                           }
+                         ]
+                       }
+                     }
+                   ]
+                 }
+               ]
+             }
+           }
+         ]
+       }
```

Comment:
Missing the purchase condition for complex criteria #3. Please update the appropriate tests as well.

Reply:
I am adding the purchase condition.